docs/configure/upgrading.mdx

```diff
@@ -36,24 +40,89 @@ The `upgrade` command will use whichever version you specify. For example:
+The upgrade script automatically runs a health check on all detected Storybook projects after the upgrade. This verifies that the upgrade was completed successfully and checks for common issues that might arise after an upgrade, such as duplicated dependencies, incompatible addons, or mismatched versions.
 
-{/* prettier-ignore-start */}
+The health check runs automatically for all detected Storybooks. You can also run it manually at any time using the `storybook doctor` command:
 
 <CodeSnippets path="storybook-doctor.md" />
 
-{/* prettier-ignore-end */}
+### Error handling and debugging
+
+If you encounter issues during the upgrade:
+
+1. A `debug-storybook.log` file will be created in the repository root containing all relevant logs
+2. For more detailed information, set the log level to `debug` using the `--loglevel debug` flag
+3. Create a GitHub issue with the logs if you need help resolving the problem
+
+## Command-line options
+
+The upgrade command supports several flags to customize the upgrade process:
+
+```bash
+storybook@latest upgrade [options]
+```
+
+### Available flags
+
+| Flag                             | Description                                                                                |
+| -------------------------------- | ------------------------------------------------------------------------------------------ |
+| `-c, --config-dir <dir-name...>` | Directory or directories to find Storybook configurations                                  |
+| `--debug`                        | Enable more logs for debugging (default: false)                                            |
+| `--disable-telemetry`            | Disable sending telemetry data                                                             |
+| `--enable-crash-reports`         | Enable sending crash reports to telemetry data                                             |
+| `-f, --force`                    | Force the upgrade, skipping autoblockers                                                   |
+| `--loglevel <level>`             | Define log level: `debug`, `error`, `info`, `silent`, `trace`, or `warn` (default: `info`) |
+| `--package-manager <manager>`    | Force package manager: `npm`, `pnpm`, `yarn1`, `yarn2`, or `bun`                           |
+| `-s, --skip-check`               | Skip postinstall version and automigration checks                                          |
+| `--write-logs`                   | Write all debug logs to a file at the end of the run                                       |
+| `-y, --yes`                      | Skip prompting the user                                                                    |
```

Comment:
@valentinpalkovic, when you can, adjust the upgrade options in the [CLI documentation](https://github.com/storybookjs/storybook/blob/next/docs/api/cli-options.mdx#upgrade) with the new ones (e.g., `--write-logs` and `--loglevel <level>`) so that we don't introduce misconceptions in the documentation